plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/data/RobotPreferencesData.java

```diff
@@ -0,0 +1,21 @@
+package edu.wpi.first.shuffleboard.plugin.base.data;
+
+import edu.wpi.first.shuffleboard.api.data.MapData;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RobotPreferencesData extends MapData {
+
+  public RobotPreferencesData(Map<String, Object> map) {
+    super(map);
+  }
+
+  @Override
+  public RobotPreferencesData put(String key, Object value) {
```

Comment:
why override this? Seems unnecessary now that the change to put was made in MapData.

Reply:
Oh, typing. Should just call super() then.

Reply:
This is the correct implementation. Calling super will return a new `MapData` instance.